common/enum-macros/src/lib.rs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//! Macro helpers for working with enums.
+
+#![forbid(unsafe_code)]
+#![warn(missing_docs)]
+
+extern crate proc_macro;
+
+mod num_variants;
+
+use syn::spanned::Spanned;
+use syn::{
+    parse_macro_input, Attribute, DeriveInput, Error, Ident, Meta, MetaList, NestedMeta, Result,
+};
+
+/// Derives an associated constant with the number of variants this enum has.
+///
+/// The default constant name is `NUM_VARIANTS`. This can be customized with `#[num_variants =
+/// "FOO")]`.
```

Comment:
For some reason the review didn't pick my comment. 

Could you update the comment to reflect the new syntax?